src/commands/repos/UpdateRepos.ts

```diff
@@ -131,31 +87,70 @@ export class UpdateRepos extends AbstractReposCommand {
+        }
+
         const pathToRepo = path.join(process.cwd(), '..', repoName);
         const wasGradleBuild = new GradleBuild(pathToRepo).containsGradleBuild();
 
         const repo = new Repository(pathToRepo);
 
-        await this.moveNodeModules(repo);
+        let targetCommit: string;
+        if (branch) {
+            if (commit) {
+                targetCommit = commit;
+            } else if (this.resetToRemote) {
+                targetCommit = `origin/${branch}`;
+            } else if (this.noFetch) {
+                let branchExists: boolean;
+                try {
+                    await repo.commitExists(branch);
+                    branchExists = true;
+                } catch (e) {
+                    branchExists = false;
+                }
+                targetCommit = branchExists ? branch : `origin/${branch}`;
+            } else {
+                // we did fetch, so the current remote branch is very likely the same as the one we may pull later
+                targetCommit = `origin/${branch}`;
+            }
+        } else {
+            targetCommit = tag;
+        }
+
+        const startingBranchHasCheckedInNodeModules = await repo.checkRepoHasPathInBranch({branch: 'HEAD', pathname: AbstractReposCommand.NODE_MODULES});
+        Global.isVerbose() && console.log(repoName, 'current branch has ', AbstractReposCommand.NODE_MODULES, 'checked in:', startingBranchHasCheckedInNodeModules);
+        const targetBranchHasCheckedInNodeModules = await repo.checkRepoHasPathInBranch({branch: targetCommit, pathname: AbstractReposCommand.NODE_MODULES});
+        Global.isVerbose() && console.log(repoName, 'target branch ', targetCommit, ' has ', AbstractReposCommand.NODE_MODULES, 'checked in:', targetBranchHasCheckedInNodeModules);
+
+        if (!startingBranchHasCheckedInNodeModules && targetBranchHasCheckedInNodeModules) {
+            console.log(`[${repoName}]: Removing untracked ${AbstractReposCommand.NODE_MODULES} folder because it is checked in in the target branch.\n` +
+                            `  This may interfere with a running cplace-asc process!`);
+            await this.removeNodeModules(repo);
+        } else if (startingBranchHasCheckedInNodeModules && targetBranchHasCheckedInNodeModules) {
+            console.log(`[${repoName}]: Folder ${AbstractReposCommand.NODE_MODULES} is checked in in the current and the target branch.\n` +
+                            `  Switching branches may interfere with a running cplace-asc process!`);
+        }
+
         if (branch) {
-            await repo.checkoutBranch(branch);
             await repo.resetHard();
+            await repo.checkoutBranch(branch);
             if (commit) {
                 await repo.checkoutCommit(commit);
             } else if (this.resetToRemote) {
                 await repo.resetHard(branch);
+            } else if (this.noFetch) {
+                // don't update to the remote branch, but stay at the current local branch
```

Comment:
This is an unrelated bugfix: Documentation says when --nofetch is set, we check out the local branch, not the remote branch.